test/integration/IntegrationTest.cpp

```diff
@@ -259,824 +292,843 @@ bool isActiveTalker(const std::vector<api::ConferenceEndpoint>& endpoints, const
     return it->isActiveTalker;
 }
 
-TEST_F(IntegrationTest, plain)
+void IntegrationTest::runTestInThread(const size_t expectedNumThreads, std::function<void()> test)
 {
-    if (__has_feature(address_sanitizer) || __has_feature(thread_sanitizer))
-    {
-        return;
-    }
-#if !ENABLE_LEGACY_API
-    return;
+#if USE_FAKENETWORK
+    // Start internet here, before runFor(...) so it goes into "nanoSleep".
+    _internet->start();
 #endif
 
-    _config.readFromString(R"({
+    std::thread runner([test] { test(); });
+
+    _timeSource.waitForThreadsToSleep(expectedNumThreads, 5 * utils::Time::sec);
+
+#if USE_FAKENETWORK
+    _timeSource.runFor(80 * utils::Time::sec);
+#endif
+
+    runner.join();
+}
+void IntegrationTest::startSimulation()
+{
+#if USE_FAKENETWORK
+    _internet->start();
+    utils::Time::nanoSleep(1 * utils::Time::sec);
+#endif
+}
+
+void IntegrationTest::finalizeSimulation()
+{
+    _internet->stop();
+    utils::Time::initialize();
+    _timeSource.shutdown();
+}
+
+TEST_F(IntegrationTest, plain)
+{
+    runTestInThread(_numWorkerThreads + 4, [this]() {
```

Comment:
We must look for a way around this.

Reply:
Please be more specific. Are you talking about number of threads we're waiting for or about lambda?